client/me/purchases/cancel-purchase/index.jsx

```diff
@@ -77,25 +77,20 @@ class CancelPurchase extends React.Component {
 			return true;
 		}
 
-		const { purchase, selectedSite } = props;
+		const { purchase } = props;
 
 		// For domain transfers, we only allow cancel if it's also refundable
 		const isDomainTransferCancelable = isRefundable( purchase ) || ! isDomainTransfer( purchase );
 
-		return selectedSite && purchase && isCancelable( purchase ) && isDomainTransferCancelable;
```

Comment:
Remove `selectedSite` dependency for `isDataValid`. Otherwise this redirects when loading disconnected site.